adapters/src/main/java/com/squareup/moshi/adapters/RuntimeJsonAdapterFactory.java

```diff
@@ -39,16 +42,25 @@
  *
  *   Moshi moshi = new Moshi.Builder()
  *       .add(RuntimeJsonAdapterFactory.of(Message.class, "type")
- *           .registerSubtype(Success.class, "success")
- *           .registerSubtype(Error.class, "error"))
+ *           .withSubtype(Success.class, "success")
+ *           .withSubtype(Error.class, "error"))
  *       .build();
  * }</pre>
  */
 // TODO(jwilson): make this class public in Moshi 1.8.
 final class RuntimeJsonAdapterFactory<T> implements JsonAdapter.Factory {
```

Comment:
public in a followup?

Reply:
Yeah, I still wanna find a way to stream out the type itself.